lib/handlers/index.js

```diff
@@ -62,6 +62,13 @@ function handle (client, packet, done) {
   if (client._keepaliveInterval > 0) {
     client._keepaliveTimer.reschedule(client._keepaliveInterval)
   }
+
+  function finish (conn, packet, done) {
+    conn.destroy()
+    var error = new Error('Invalid protocol')
+    error.info = packet
+    done(error)
+  }
```

Comment:
please avoid creating a closure here, they use memory and this is a hot path.

Reply:
sure, I haven't noticed that. :P